app/_hub/kong-inc/ldap-auth-advanced/index.md

```diff
@@ -78,85 +82,99 @@ params:
     - name: verify_ldap_host
       required: true
       default: "`false`"
       value_in_examples: false
+      datatype: boolean
       description: |
         Set to `true` to authenticate LDAP server. The server certificate will be verified according to the CA certificates specified by the `lua_ssl_trusted_certificate` directive.
     - name: attribute
       required: true
       default:
       value_in_examples: cn
+      datatype: string
       description: |
         Attribute to be used to search the user; e.g., "cn".
     - name: cache_ttl
       required: true
       default: "`60`"
       value_in_examples: 60
+      datatype: number
       description: |
         Cache expiry time in seconds.
     - name: timeout
       required: false
       default: "`10000`"
       value_in_examples:
+      datatype: number
       description: |
         An optional timeout in milliseconds when waiting for connection with LDAP server.
     - name: keepalive
       required: false
       default: "`10000`"
       value_in_examples:
+      datatype: number
       description: |
         An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
     - name: anonymous
       required: false
       default:
       value_in_examples:
+      datatype: string
       description: |
         An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
     - name: header_type
       required: false
       default: "`ldap`"
       value_in_examples: ldap
+      datatype: string
       description: |
         An optional string to use as part of the Authorization header. By default, a valid Authorization header looks like this: `Authorization: ldap base64(username:password)`. If `header_type` is set to "basic", then the Authorization header would be `Authorization: basic base64(username:password)`. Note that `header_type` can take any string, not just `"ldap"` and `"basic"`.
     - name: consumer_optional
       required: false
       default: "`false`"
       value_in_examples:
+      datatype: boolean
       description: |
         Whether consumer mapping is optional. If `consumer_optional=true`, the plugin will not attempt to associate a consumer with the LDAP authenticated user. If `consumer_optional=false`, LDAP authenticated users can still access upstream resources. To prevent access from LDAP users that are not associated with consumers, set `consumer_optional=false`, set the `anonymous` field to an existing `consumer_id`, then use the Request Termination plugin to deny any requests from the anonymous consumer.
     - name: consumer_by
       required: false
       default: '`[ "username", "custom_id" ]`'
       value_in_examples:
+      datatype:
```

Comment:
?

array of string elements

Reply:
can be one of "username", "custom_id"

Reply:
> can be one of "username", "custom_id"

The schema gives both as defaults, description says and/or, yet schema says one of? So it must be one or both of those options.

>   { consumer_by = {
          type = "array",
          elements = { type = "string", one_of = { "username", "custom_id" }},
          required = false,
          default = { "username", "custom_id" },

Reply:
Please review again. commit afbee5b.